pkgs/applications/editors/vscode/extensions/shader-slang.slang-language-extension/default.nix

```
@@ -0,0 +1,79 @@
+{
+  lib,
+  shader-slang,
+  stdenv,
+  vscode-utils,
+}:
+
+vscode-utils.buildVscodeMarketplaceExtension {
+  mktplcRef = {
+    name = "slang-language-extension";
+    publisher = "shader-slang";
+    version = "2.0.1";
+    hash = "sha256-cMBFksLcEi3OmRpyuk/hMNo5HqsfGvWhheiLg6Dusn0=";
+  };
+
+  postInstall =
+    let
+      # https://github.com/shader-slang/slang-vscode-extension/blob/v2.0.1/build_scripts/build-package.sh#L40-L43
+      binaries = {
+        "x86_64-linux" = {
+          name = "linux-x64";
+          files = [
+            "lib/libslang.so"
+            "lib/libslang-glsl-module.so"
+            "bin/slangd"
+          ];
+        };
+        "aarch64-linux" = {
+          name = "linux-arm64";
+          files = [
+            "lib/libslang.so"
+            "lib/libslang-glsl-module.so"
+            "bin/slangd"
+          ];
+        };
+        "x86_64-darwin" = {
+          name = "darwin-x64";
+          files = [
+            "lib/libslang.dylib"
+            "lib/libslang-glsl-module.dylib"
+            "bin/slangd"
+          ];
+        };
+        "aarch64-darwin" = {
+          name = "darwin-arm64";
+          files = [
+            "lib/libslang.dylib"
+            "lib/libslang-glsl-module.dylib"
+            "bin/slangd"
+          ];
+        };
+      };
```

Just pull the extensions and make a mapping of the host to the path. Then immediately index it. 

You optional was just wrong, and would have created an incorrect package on unsupported platforms. This makes a nice loud error. 
```suggestion
            # https://github.com/shader-slang/slang-vscode-extension/blob/v2.0.1/build_scripts/build-package.sh#L40-L43
      name = {
        "x86_64-linux" = "linux-x64";
        "aarch64-linux" = "linux-arm64";
        "x86_64-darwin" = "darwin-x64";
        "aarch64-darwin" = "darwin-arm64";
      }.${stdenv.hostPlatform.system};

      files = [
        "lib/libslang${stdenv.hostPlatform.extensions.library}"
        "lib/libslang-glsl-module${stdenv.hostPlatform.extensions.library}"
        "bin/slangd${stdenv.hostPlatform.extensions.executable}"
      ];
```